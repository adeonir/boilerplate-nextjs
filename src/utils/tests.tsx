import { cleanup, render } from '@testing-library/react'
import type { ReactElement } from 'react'
import { afterEach } from 'vitest'

afterEach(() => {
  cleanup()
})

function customRender(ui: ReactElement, options = {}) {
  return render(ui, {
    wrapper: ({ children }) => children,
    ...options,
  })
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
export { customRender as render }
