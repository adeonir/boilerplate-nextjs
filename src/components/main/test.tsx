import { render, screen } from '~/utils/tests'

import { Main } from '.'

const props = {
  title: 'Next.js Boilerplate',
  description: 'Next.js, TypeScript, Tailwind, Storybook, Vitest, Cypress and Testing-Library',
}

describe('<Main />', () => {
  it('should render the headings', () => {
    const { container } = render(<Main {...props} />)

    expect(screen.getByRole('heading', { name: props.title })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: props.description })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
