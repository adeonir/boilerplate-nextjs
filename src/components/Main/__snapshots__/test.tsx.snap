// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #051330;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4rem;
  text-align: center;
}

.c1 {
  margin-bottom: 3rem;
  width: 25rem;
}

.c2 {
  font-size: 2.5rem;
  margin-bottom: 1rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 300;
}

.c4 {
  margin-top: 4rem;
  width: min(40rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado"
    class="c1"
    src="/assets/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    NextJS, ReactJS, TypeScript e Styled-Components
  </h2>
  <img
    alt="Um desenvolvedor caminhando e uma tela com código e também escrito a palavra Next.JS"
    class="c4"
    src="/assets/hero-illustration.svg"
  />
</main>
`
